test/e2e/dra/dra.go

```diff
@@ -491,3 +491,10 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 		})
 	})
 })
+
+func TestInlinePod(ctx context.Context, f *framework.Framework, b *Builder) {
+	parameters := b.parameters()
+	pod, template := b.podInline(resourcev1alpha2.AllocationModeWaitForFirstConsumer)
+	b.create(ctx, parameters, template)
+	b.testPod(ctx, f.ClientSet, pod)
```

Comment:
@pohly this test fails as pod doesn't get scheduled to the node:
```
  [FAILED] start pod: Timed out after 300.001s.
  The function passed to Eventually returned the following error:
      <framework.transientError>: 
      pods "tester-1" not found
      {
          error: <*errors.StatusError | 0xc0007dbae0>{
              ErrStatus: {
                  TypeMeta: {Kind: "", APIVersion: ""},
                  ListMeta: {
                      SelfLink: "",
                      ResourceVersion: "",
                      Continue: "",
                      RemainingItemCount: nil,
                  },
                  Status: "Failure",
                  Message: "pods \"tester-1\" not found",
                  Reason: "NotFound",
                  Details: {Name: "tester-1", Group: "", Kind: "pods", UID: "", Causes: nil, RetryAfterSeconds: 0},
                  Code: 404,
              },
          },
      }
  In [It] at: test/e2e/dra/builder.go:270 @ 05/16/23 14:21:48.28
```

Reply:
@pohly In other e2e_node tests scheduling is bypassed by [explicitly assigning pod to the node](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/pod/pod_client.go#L203)

Reply:
The scheduler is not running. Doing the normal scheduling + allocation will not work.

Why do you want to run this test as part of `e2e_node`?

My expectation was that we would have dedicated tests for kubelet in `e2e_node` and not try to run the "normal" ones.

Reply:
This is just a simplest example I came up with. Generally I want to test a Kubelet-specifics like restarts, checkpointing, grpc timeouts etc in the working environment, i.e. when pods are scheduled to the node normally. Should I consider another approach?

Reply:
The test code has to work without kube-scheduler and without kube-controller-manager. That means you have to create Pods and ResourceClaims already with the fields as set by those components (pod.spec.selectedNode, claim.status, etc.).

Reply:
Thank you for the explanation. Now I understand how it's designed to work.

Reply:
@pohly can you suggest how to debug this? As I don't know which fields to update the result is expected: everything resource claim related is only present in the pod `managedFields`. Pod runs, but it doesn't contain any claim references. I'm guessing that update operation hasn't succeeded as some fields are either not set at all or set incorrectly:
```
{
  "metadata": {
    "name": "dra-node-tester-2154380e-ead9-4bdd-8b2a-8dd6cc91b70e",
    "namespace": "dra-node-9394",
    "uid": "9448f035-6ea4-48be-9f5b-b57f4c30587b",
    "resourceVersion": "96",
    "creationTimestamp": "2023-05-28T17:32:27Z",
    "annotations": {
      "kubernetes.io/config.seen": "2023-05-28T20:32:27.400355816+03:00",
      "kubernetes.io/config.source": "api"
    },
    "managedFields": [
      {
        "manager": "e2e_node.test",
        "operation": "Update",
        "apiVersion": "v1",
        "time": "2023-05-28T17:32:27Z",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:spec": {
            "f:containers": {
              "k:{\"name\":\"testcontainer\"}": {
                ".": {},
                "f:command": {},
                "f:image": {},
                "f:imagePullPolicy": {},
                "f:name": {},
                "f:resources": {
                  ".": {},
                  "f:claims": {
                    ".": {},
                    "k:{\"name\":\"resource-claim\"}": {
                      ".": {},
                      "f:name": {}
                    }
                  }
                },
                "f:terminationMessagePath": {},
                "f:terminationMessagePolicy": {}
              }
            },
            "f:dnsPolicy": {},
            "f:enableServiceLinks": {},
            "f:nodeName": {},
            "f:resourceClaims": {
              ".": {},
              "k:{\"name\":\"resource-claim\"}": {
                ".": {},
                "f:name": {},
                "f:source": {
                  ".": {},
                  "f:resourceClaimName": {}
                }
              }
            },
            "f:restartPolicy": {},
            "f:schedulerName": {},
            "f:securityContext": {},
            "f:terminationGracePeriodSeconds": {}
          }
        }
      }
    ]
  },
  "spec": {
    "containers": [
      {
        "name": "testcontainer",
        "image": "registry.k8s.io/e2e-test-images/busybox:1.29-4",
        "command": [
          "/bin/env"
        ],
        "resources": {},
        "terminationMessagePath": "/dev/termination-log",
        "terminationMessagePolicy": "File",
        "imagePullPolicy": "IfNotPresent"
      }
    ],
    "restartPolicy": "Never",
    "terminationGracePeriodSeconds": 30,
    "dnsPolicy": "Default",
    "nodeName": "devel",
    "securityContext": {},
    "schedulerName": "default-scheduler",
    "tolerations": [
      {
        "key": "node.kubernetes.io/not-ready",
        "operator": "Exists",
        "effect": "NoExecute",
        "tolerationSeconds": 300
      },
      {
        "key": "node.kubernetes.io/unreachable",
        "operator": "Exists",
        "effect": "NoExecute",
        "tolerationSeconds": 300
      }
    ],
    "priority": 0,
    "enableServiceLinks": true,
    "preemptionPolicy": "PreemptLowerPriority"
  },
  "status": {
    "phase": "Pending",
    "qosClass": "BestEffort"
  }
}
```